src/ruby.rs

```diff
@@ -44,7 +44,148 @@ impl_from_arity!(from_arity_13, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE, VALUE,
+    pub static T_ARRAY: ruby_value_type;
+
+    #[link_name = "RS_T_HASH"]
+    pub static RB_T_HASH: ruby_value_type;
+    #[link_name = "RS_T_HASH"]
+    pub static T_HASH: ruby_value_type;
+
+    #[link_name = "RS_T_STRUCT"]
+    pub static RB_T_STRUCT: ruby_value_type;
+    #[link_name = "RS_T_STRUCT"]
+    pub static T_STRUCT: ruby_value_type;
+
+    #[link_name = "RS_T_BIGNUM"]
+    pub static RB_T_BIGNUM: ruby_value_type;
+    #[link_name = "RS_T_BIGNUM"]
+    pub static T_BIGNUM: ruby_value_type;
+
+    #[link_name = "RS_T_FILE"]
+    pub static RB_T_FILE: ruby_value_type;
+    #[link_name = "RS_T_FILE"]
+    pub static T_FILE: ruby_value_type;
+
+    #[link_name = "RS_T_DATA"]
+    pub static RB_T_DATA: ruby_value_type;
+    #[link_name = "RS_T_DATA"]
+    pub static T_DATA: ruby_value_type;
+
+    #[link_name = "RS_T_MATCH"]
+    pub static RB_T_MATCH: ruby_value_type;
+    #[link_name = "RS_T_MATCH"]
+    pub static T_MATCH: ruby_value_type;
+
+    #[link_name = "RS_T_COMPLEX"]
+    pub static RB_T_COMPLEX: ruby_value_type;
+    #[link_name = "RS_T_COMPLEX"]
+    pub static T_COMPLEX: ruby_value_type;
+
+    #[link_name = "RS_T_RATIONAL"]
+    pub static RB_T_RATIONAL: ruby_value_type;
+    #[link_name = "RS_T_RATIONAL"]
+    pub static T_RATIONAL: ruby_value_type;
+
+    #[link_name = "RS_T_NIL"]
+    pub static RB_T_NIL: ruby_value_type;
+    #[link_name = "RS_T_NIL"]
+    pub static T_NIL: ruby_value_type;
+
+    #[link_name = "RS_T_TRUE"]
+    pub static RB_T_TRUE: ruby_value_type;
+    #[link_name = "RS_T_TRUE"]
+    pub static T_TRUE: ruby_value_type;
+
+    #[link_name = "RS_T_FALSE"]
+    pub static RB_T_FALSE: ruby_value_type;
+    #[link_name = "RS_T_FALSE"]
+    pub static T_FALSE: ruby_value_type;
+
+    #[link_name = "RS_T_SYMBOL"]
+    pub static RB_T_SYMBOL: ruby_value_type;
+    #[link_name = "RS_T_SYMBOL"]
+    pub static T_SYMBOL: ruby_value_type;
+
+    #[link_name = "RS_T_FIXNUM"]
+    pub static RB_T_FIXNUM: ruby_value_type;
+    #[link_name = "RS_T_FIXNUM"]
+    pub static T_FIXNUM: ruby_value_type;
+
+    #[link_name = "RS_T_UNDEF"]
+    pub static RB_T_UNDEF: ruby_value_type;
+    #[link_name = "RS_T_UNDEF"]
+    pub static T_UNDEF: ruby_value_type;
+
+    #[link_name = "RS_T_IMEMO"]
+    pub static RB_T_IMEMO: ruby_value_type;
+    #[link_name = "RS_T_IMEMO"]
+    pub static T_IMEMO: ruby_value_type;
+
+    #[link_name = "RS_T_NODE"]
+    pub static RB_T_NODE: ruby_value_type;
+    #[link_name = "RS_T_NODE"]
+    pub static T_NODE: ruby_value_type;
+
+    #[link_name = "RS_T_ICLASS"]
+    pub static RB_T_ICLASS: ruby_value_type;
+    #[link_name = "RS_T_ICLASS"]
+    pub static T_ICLASS: ruby_value_type;
+
+    #[link_name = "RS_T_ZOMBIE"]
+    pub static RB_T_ZOMBIE: ruby_value_type;
+    #[link_name = "RS_T_ZOMBIE"]
+    pub static T_ZOMBIE: ruby_value_type;
+
+    #[link_name = "RS_T_MASK"]
+    pub static RB_T_MASK: ruby_value_type;
+    #[link_name = "RS_T_MASK"]
+    pub static T_MASK: ruby_value_type;
```

Comment:
This one is a little bit suspicious. It is different from the rest in that it is not a thing that you would generally passing to `RB_TYPE_P` (or get back from `RB_TYPE`). It is the mask pattern (`0b11111`) that helps you mask out the type bits from the `RBasic` flags. There may eventually be a type that happens to have this same value, but it would probably have another name (so that `flags & T_MASK == T_SOME_TYPE`). For that reason it seems a bit suspicious to type this as a `ruby_value_type`. I suggest simply leaving this out for now.

Reply:
It’s part of the enum in Ruby, which is why I copied it over. I’m fine with taking it out, thought.